### ISSUE TITLE
Accessibility Issues with User Feedback Widget

### ISSUE THREAD

**Issue by devinsm** [getsentry/sentry#23288]
- [x] Review the documentation: https://docs.sentry.io/
- [x] Search for existing issues: https://github.com/getsentry/sentry-javascript/issues
- [x] Use the latest release: https://github.com/getsentry/sentry-javascript/releases
- [x] Provide a link to the affected event from your Sentry account <- Not applicable

## Package + Version

- [x] `@sentry/browser`
- [ ] `@sentry/node`
- [ ] `raven-js`
- [ ] `raven-node` _(raven for node)_
- [ ] other:

### Version:

```
5.29.2
```

## Description

The dialog opened by Sentry.showReportDialog is not accessible. See the [WAI-ARIA Authoring Practices guidance on accessible modals](https://www.w3.org/TR/wai-aria-practices-1.1/#dialog_modal).

Some specific issues which need to be fixed:

- `Tab` and `Shift` + `Tab` should not move focus to elements outside the modal (they currently can)
- When the modal loads, the first input of the form should receive focus (currently nothing in the modal is focused when the modal loads)
- The "Close" button is rendered using an anchor tag without an `href` attribute. As a result it is not in the tab sequence and keyboard users are not able to use it. To fix this, a `button` element should be used instead. Since the element does not cause navigation, the `button` element will also have the proper semantics and will improve the experience for users of assistive technology.
- The outermost div of the dialog needs `role` set to `dialog`
- The outermost div of the dialog needs `aria-modal` set to `true`
- The outermost div of the dialog needs `aria-labelledby` set to the `id` of the modal's h2

**Comment by kamilogorek** [comment]
Transferred the issue, as User Feedback Dialog lives in this repository.
